### DISCUSSION TITLE
Language not being detected by linguist

### DISCUSSION THREAD

**Discussion by mrjithin** [github-linguist/linguist#5467] [Classification] [answered]
Consider my repo which is not yet completed: [username-generator](https://github.com/jit-hin/username-generator/).

The language used is *not detected* by linguist. I know that it will take time but after so many weeks also my old repo which was identical to the new one, it's language wasn't detected. 

I wanted to know if I am using minified JavaScript as I have read the conditions of linguist. Please do help me.

*Thanks!*

**Comment by lildude** [comment, marked as answer]
Everything in `node_modules` is ignored as Linguist considers the location for vendored code, leaving just your `app.js` file.

Linguist is detecting this as generated:

```console
$ github-linguist username-generator/app.js
app.js: 25 lines (24 sloc)
  type:      Text
  mime type: application/javascript
  language:  JavaScript
  appears to be generated source code
$
```

This will be because the file is short at only 25 lines, but each line is on average longer than 110 chars thanks to all those block in your banner output (in the `console.log()` statements) so is being detected as generated here:

https://github.com/github/linguist/blob/de19dc13e02a3e8d2bbf7b238c6bd8b0e5f35581/lib/linguist/generated.rb#L137-L151

You will need to implemented an override to tell Linguist this file is definitely not generated or change the content.
